python-frontend/src/main/java/org/sonar/python/types/TypeShed.java

```diff
@@ -214,6 +214,9 @@ public static boolean isValidForProjectPythonVersion(List<String> validForPython
     if (validForPythonVersions.isEmpty()) {
       return true;
     }
+    if (supportedPythonVersions.stream().allMatch("312"::equals) && validForPythonVersions.contains("311")) {
+      return true;
```

Comment:
I had the impression you pushed this, but it seems not: there should be a `TODO` comment here with link to SONARPY-1522.

Reply:
Addressed in 5b320b2eae5bfcb712c343ad32d7b891c30bdb1b.